src/__tests__/reducer.test.ts

```diff
@@ -0,0 +1,235 @@
+import { auth0Loaded, handleRedirectCallbackAction, loginWithPopupAction } from "../actions";
+import { auth0ProviderStateReducer } from "../reducer";
+import {
+  HandlingRedirectState,
+  isAuthenticatedState,
+  isLoadedState,
+  isUnauthenticatedState,
+  LoadingState,
+  TokenUser,
+  UnauthenticatedState,
+} from "../types";
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const createFakeAuth0Client = (): any => {
+  return {
+    loginWithRedirect: jest.fn(),
+    getTokenWithPopup: jest.fn(),
+    getTokenSilently: jest.fn(),
+    getIdTokenClaims: jest.fn(),
+    logout: jest.fn(),
+  };
+};
+
+const createInitialState = (): LoadingState => ({
+  loading: true,
+  isAuthenticated: false,
+  popupOpen: false,
+  user: undefined,
+  auth0Client: undefined,
+  handlingRedirect: false,
+});
+
+const createFakeUnauthenticatedState = (): UnauthenticatedState => ({
```

Comment:
This is so simple that it may not be worth it but I've started playing with `test-data-bot` https://github.com/jackfranklin/test-data-bot for mock data.  It uses faker under the hood which is also pretty nice.

Reply:
I created a separate issue to take a look at this one